src/main/java/dev/openfeature/javasdk/Structure.java

```diff
@@ -0,0 +1,118 @@
+package dev.openfeature.javasdk;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import dev.openfeature.javasdk.internal.Pair;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@ToString
+@EqualsAndHashCode
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class Structure {
```

Comment:
If our backend returns this:

```json
{
  "foo": null
}
```

What type will that serialize into on a structure?

Reply:
Provider won't know what type to use. I think the way we work around this is returning `null` for any unknown values. This means if you come across a `null`, you can just not call `add` at all.

Reply:
> If we don't call add, then foo won't be in the keySet. So... let's just pick something like boolean and shove all nulls in there as a typecasted thing. Users will get nulls out when they ask for them.

OK that makes sense. That's really something for the provider to do though, right? I guess we could add tests for this, but there isn't much for us to do in the SDK to support this, right?

Reply:
That's right.

Reply:
I'll add a specific test for this.

Reply:
I've added such a test.